mesheryctl/cmd/perf.go

```diff
@@ -0,0 +1,136 @@
+
+// StringWithCharset generates a random string with a given length
+func StringWithCharset(length int) string {
+	const charset = "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = charset[seededRand.Intn(len(charset))]
+	}
+	return string(b)
+}
+
+// perfCmd represents the Performance command
+var perfCmd = &cobra.Command{
+	Use:   "perf",
+	Short: "Performance testing and benchmarking",
+	Long:  `Performance testing and benchmarking.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		//Check prerequisite
+		preReqCheck()
+
+		println("Test name used : ", testName)
+
+		const mesheryURL string = "http://localhost:9081/api/load-test-smps?"
+		postData := ""
+
+		startTime := time.Now()
+		duration, err := time.ParseDuration(testDuration)
+		if err != nil {
+			println("Error: Test duration invalid")
+			return
+		}
+		endTime := startTime.Add(duration)
+
+		postData = postData + "start_time: " + startTime.Format(time.RFC3339)
+		postData = postData + "\nend_time: " + endTime.Format(time.RFC3339)
+
+		if len(testURL) > 0 {
+			postData = postData + "\nendpoint_url: " + testURL
+		} else {
+			println("Error: Please enter a TestURL")
+			return
+		}
+
+		postData = postData + "\nclient:"
+		postData = postData + "\n connections: " + parallelRequests
+		postData = postData + "\n rps: " + qps
+
+		req, err := http.NewRequest("POST", mesheryURL, bytes.NewBuffer([]byte(postData)))
+		if err != nil {
+			println("Error in building the request")
+			return
+		}
+		cookieConf := strings.SplitN(testCookie, "=", 2)
+		cookieName := cookieConf[0]
+		cookieValue := cookieConf[1]
+		req.AddCookie(&http.Cookie{Name: cookieName, Value: cookieValue})
+		q := req.URL.Query()
+		q.Add("name", testName)
+		q.Add("loadGenerator", loadGenerator)
+		if len(testMesh) > 0 {
+			q.Add("mesh", testMesh)
+		}
+		req.URL.RawQuery = q.Encode()
+
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			return
+		}
+
+		buf := make([]byte, 4)
+		for {
+			n, err := resp.Body.Read(buf)
+			fmt.Print(string(buf[:n]))
+			if err == io.EOF {
+				break
+			}
+		}
+		println("\nTest Completed Successfully!")
+	},
+}
+
+func init() {
+	perfCmd.Flags().StringVar(&testURL, "url", "", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&testName, "name", StringWithCharset(8), "DESCRIPTION")
+	perfCmd.Flags().StringVar(&testMesh, "mesh", "", "DESCRIPTION")
+	// perfCmd.Flags().StringVar(&testFile, "file", "", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&qps, "qps", "1", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&parallelRequests, "parallel-requests", "1", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&testDuration, "duration", "10s", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&testCookie, "cookie", "meshery-provider=Local (ephemeral session) (free use)", "DESCRIPTION")
+	perfCmd.Flags().StringVar(&loadGenerator, "load-generator", "fortio", "DESCRIPTION")
```

Comment:
Might want to lookup the available load-generators (as this is an area of extensibility for Meshery) and then validate that the string provided matched an available generator.